src/retrieval.rs

```diff
@@ -129,8 +130,108 @@ pub fn fetch_entries<
     }
 }
 
+
+pub fn err(reason: &str) -> WasmError {
+    WasmError::Guest(String::from(reason))
+}
+
+// what is this function doing?
+pub fn get_last_component_string(path_tag: LinkTag) -> ExternResult<String> {
+    let hour_path = Path::try_from(&path_tag)?;
+    let hour_components: Vec<hdk::hash_path::path::Component> = hour_path.into();
+
+    let hour_bytes: &hdk::hash_path::path::Component = hour_components.last().ok_or(err("Invalid path"))?;
+    let hour_str: String = hour_bytes.try_into()?;
+
+    Ok(hour_str)
+}
+
+pub fn fetch_entries_by_time<
+    EntryType: TryFrom<SerializedBytes, Error = SerializedBytesError>,
+>(time: FetchEntriesTime, base_component: String) -> Result<Vec<WireElement<EntryType>>, WasmError> {
+    let entries = match time.hour {
+        None => fetch_entries_by_day(time, base_component),
+        Some(h) => fetch_entries_by_hour(time.year, time.month, time.day, h, base_component),
+    }?;
+
+    Ok(entries)
```

Comment:
```suggestion
    Ok(match time.hour {
        None => fetch_entries_by_day(time, base_component),
        Some(h) => fetch_entries_by_hour(time.year, time.month, time.day, h, base_component),
    }?)
```

we can do this because the `?` still will early exit the function (order-of-execution) and produce Err result appropriately